src/sys/select.rs

```diff
@@ -227,6 +300,29 @@ mod tests {
         assert!(!fd_set.contains(r2));
     }
 
+    #[test]
+    #[cfg_attr(any(target_arch = "powerpc", target_arch = "mips"), ignore)]
+    fn test_pselect() {
+        let (r1, w1) = pipe().unwrap();
+        write(w1, b"hi!").unwrap();
+        let (r2, _w2) = pipe().unwrap();
+
+        let mut fd_set = FdSet::new();
+        fd_set.insert(r1);
+        fd_set.insert(r2);
+
+        let timeout = TimeSpec::seconds(10);
+        let sigmask = SigSet::empty();
+        assert_eq!(1, pselect(None,
+                              &mut fd_set,
+                              None,
+                              None,
+                              &timeout,
+                              &sigmask).unwrap());
```

Comment:
Altering the signal mask has side effects which may interfere with other concurrent tests.  Nix has a way of dealing with that: you should this test into test/sys/test_select.rs and then add this line to the top:
```rust
let _mtx = ::SIGNAL_MTX.lock().expect("Mutex got poisoned by another test");
```
Since the other test does not modify the sigmask, you don't need the mutex.  It's up to you whether or not to move it.

Reply:
Thanks - I'll move both.